src/app/Fake.Runtime/CompileRunner.fs

```diff
@@ -92,6 +92,19 @@ let tryRunCached (c:CoreCacheInfo) (context:FakeContext) : RunResult =
     | None -> RunResult.SuccessRun c.Warnings
     | Some e -> RunResult.RuntimeError e
 
+/// options that must be added to compilations to circumvent compilation errors from the DependencyManager preview feature
+let fcsDependencyManagerOptions =
+    let dummyPaketDependencyManagerOption =
+        match typeof<Marker>.Assembly.Location with
+        | "" -> []
+        | s ->
+            let currentDir = Path.GetDirectoryName s
+            [ sprintf "--compilertool:%s" currentDir ]
+
+    "--langversion:preview"  // needed because of a design choice(bug?) in FCS that parses dependency managers regardless of langversion
+    :: dummyPaketDependencyManagerOption // needed to handle and swallow the `paket` dependency manager type
+    @ [ "--nowarn:3186" ] // needed because the paket dependencymanager build right now throws some kind of pickling warning.
```

Comment:
are there other places I'm missing to provide this nowarn? the tests are failing because of that warning I'm trying to suppress here.

Reply:
To be honest: I guess that `--nowarn` just doesn't work for this warning, so you probably just found another bug (welcome to my world, going down the rabit hole)

Reply:
It worked when I added it to FSAC to sidestep this same exact error, though. So I am confused.

Reply:
Good point. In any case, I don't think there are any other options. On the other hand this is probably reproducible and debuggable so I'd suggest setting a breakpoint here and see if FCS outputs the warning.

IIRC we collect the warnings and report them later so I think we can filter manually as well (as an alternative approach).

BTW have you already reported this strange warning?

Reply:
I was going to try to make an independent repro to give to the team, something self contained.

Reply:
I already created the issue in https://github.com/dotnet/fsharp/issues/8678 in order to link in the commit messages. Feel free to add the sample there.

It looks like `--nowarn` is not evaluated in FCS (which I guess makes sense as you easily can filter the returned list yourself). So I changed to this: https://github.com/fsharp/FAKE/pull/2466/commits/9f01307c3d83a770dc1df67537c5334b22207d65